AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/TransformerV2/DataStoreConnectionScenario6V2Tests.swift

```diff
@@ -0,0 +1,526 @@
+                    }
+                }
+                if let commentEvent = try? mutationEvent.decodeModel() as? Comment6V2, commentEvent.id == comment.id {
+                    if mutationEvent.mutationType == GraphQLMutationType.create.rawValue {
+                        XCTAssertEqual(commentEvent.content, comment.content)
+                        XCTAssertEqual(mutationEvent.version, 1)
+                        createReceived.fulfill()
+                        return
+                    }
+                }
+        }
+        guard try HubListenerTestUtilities.waitForListener(with: hubListener, timeout: 5.0) else {
+            XCTFail("Listener not registered for hub")
+            return
+        }
+        let getBlogCompleted = expectation(description: "get blog complete")
+        var resultPosts: List<Post6V2>?
+        Amplify.DataStore.query(Blog6V2.self, byId: blog.id) { result in
+            switch result {
+            case .success(let queriedBlogOptional):
+                guard let queriedBlog = queriedBlogOptional else {
+                    XCTFail("Could not get blog")
+                    return
+                }
+                XCTAssertEqual(queriedBlog.id, blog.id)
+                resultPosts = queriedBlog.posts
+                getBlogCompleted.fulfill()
+            case .failure(let response): XCTFail("Failed with: \(response)")
+            }
+        }
+        wait(for: [getBlogCompleted], timeout: TestCommonConstants.networkTimeout)
+        guard let posts = resultPosts else {
+            XCTFail("Could not get posts")
+            return
+        }
+        XCTAssertEqual(posts.count, 1)
+        guard let fetchedPost = posts.first(where: { (postFetched) -> Bool in
+            postFetched.id == post.id
+        }), let comments = fetchedPost.comments else {
+            XCTFail("Could not set up - failed to get a post and its comments")
+            return
+        }
+        XCTAssertEqual(comments.count, 1)
+        XCTAssertTrue(comments.contains(where: { (commentFetched) -> Bool in
+            commentFetched.id == comment.id
+        }))
+        wait(for: [createReceived], timeout: TestCommonConstants.networkTimeout)
+    }
+
+    func testCascadeDeleteBlogDeletesPostAndComments() throws {
+        try startAmplifyAndWaitForSync()
+        guard let blog = saveBlog(name: "name"),
+              let post = savePost(title: "title", blog: blog),
+              let comment = saveComment(post: post, content: "content") else {
+            XCTFail("Could not create blog, posts, and comments")
+            return
+        }
+        let createReceived = expectation(description: "Create notification received")
+        createReceived.expectedFulfillmentCount = 3 // 3 models (1 blog and 1 post and 1 comment)
+        let deleteReceived = expectation(description: "Delete notification received")
+        deleteReceived.expectedFulfillmentCount = 3 // 3 models due to cascade delete behavior
```

Comment:
this test waits until data has been reconciled from cloud and then performs a local delete of the blog. the expected behavior is deleting the blog cascades to the post and comment. 

![image](https://user-images.githubusercontent.com/1365977/144919481-aca41a60-ce59-4aef-860b-77fbb8ddd3eb.png)
![image](https://user-images.githubusercontent.com/1365977/144919508-a869bf62-46f0-4c6d-bf81-892e3f666fec.png)
![image](https://user-images.githubusercontent.com/1365977/144919528-0677fc85-c112-4c29-84e6-0364a3ccf3f5.png)